### PR TITLE
Introduce onboarding message for first-time commenters

### PR DESCRIPTION
**Pull request by nicl**
# My Awesome Pull Request
## What does this change?

Introduces an onboarding stage for first-time commenters. When pressing 'post' the user is shown a welcome message linking to the community standards and showing a preview of the comment.
## What is the value of this and can you measure success?

The hope is that the onboarding will educate users about our community standards and reduce the number of abusive comments.
## Does this affect other platforms - Amp, Apps, etc?

No.
## Screenshots

screen shot 2016-04-07 at 18 27 22
## Request for comment

I've gone over this with @OliverJAsh in the initial stages (all faults entirely my own yada yada) but I'm fumbling around here so all feedback welcome!

cc @guardian/discussion

### CODE REVIEW

File: common/app/views/fragments/commentBox.scala.html [hunk 1]
```diff
@@ -21,13 +21,24 @@
         </div>
     </div>
 
+    <div class="d-comment-box__onboarding">
+        <h3>Welcome, you’re about to make your first comment!</h3>
```
Comment (mchv) [inline common/app/views/fragments/commentBox.scala.html]: Very small suggestion.  We may add the user firstname to have a more personal message, like `Welcome John`.

Comment (nicl) [inline common/app/views/fragments/commentBox.scala.html]: That was on the original spec in fact! So definitely a good idea. I'll look to re-add it.

File: static/src/javascripts/projects/common/modules/discussion/comment-box.js [hunk 2]
```diff
@@ -145,6 +146,25 @@ CommentBox.prototype.prerender = function() {
     }
 };
 
+CommentBox.prototype.showOnboarding = function(e) {
+    e.preventDefault();
+
+    // Check if new commenter as they may have already commented on this article
+    if (this.hasState('onboarding-visible') || !this.options.newCommenter) {
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: It looks like we only call this method iff they are a new commenter?

Comment (nicl) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: @OliverJAsh the imagined scenario is a user that comments once (for the first time) and then comments again without reloading the page.

Comment (nicl) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: To expand on this, once they have commented we set newCommenter to false (see ln155 just below) but we don't rebind the events. Maybe we should? I wasn't sure about rebinding outside of the ready function.

File: common/app/views/fragments/commentBox.scala.html (hunk 1 above)
Comment (nicl) [inline common/app/views/fragments/commentBox.scala.html]: @mchv this is now done!

screen shot 2016-04-11 at 11 08 24

**Comment by nicl** [comment]
@OliverJAsh thanks for looking over it. Do you have any more thoughts on this?

File: static/src/javascripts/projects/common/modules/discussion/comment-box.js (hunk 2 above)
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: I see! You could unbind the events, but I think this looks fine.

File: static/src/javascripts/projects/common/modules/discussion/comment-box.js [hunk 3]
```diff
@@ -379,13 +416,12 @@ CommentBox.prototype.verificationEmailFail = function() {
...
  * @param {Event=} e (optional)
  */
-CommentBox.prototype.previewComment = function(e) {
+CommentBox.prototype.previewComment = function(callback) {
     var self = this,
         comment = {
             body: this.getElem('body').value
         };
 
-    e.preventDefault();
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: How come this was removed?

**Comment by OliverJAsh** [comment]
:+1: Well done for fighting the complexity here!

File: static/src/javascripts/projects/common/modules/discussion/comment-box.js (hunk 3 above)
Comment (nicl) [inline static/src/javascripts/projects/common/modules/discussion/comment-box.js]: It didn't seem to do anything!

**Pull request merged**